runtime/vam/expr/setexpr.go

```diff
@@ -0,0 +1,43 @@
+package expr
+
+import (
+	"github.com/brimdata/super"
+	"github.com/brimdata/super/vector"
+	"github.com/brimdata/super/zcode"
+)
+
+type setExpr struct {
+	sctx  *super.Context
+	elems []ListElem
+}
+
+func NewSetExpr(sctx *super.Context, elems []ListElem) Evaluator {
+	return &setExpr{sctx, elems}
+}
+
+func (s *setExpr) Eval(this vector.Any) vector.Any {
+	var vecs []vector.Any
+	for _, e := range s.elems {
+		if e.Spread != nil {
+			vecs = append(vecs, e.Spread.Eval(this))
+		} else {
+			vecs = append(vecs, e.Value.Eval(this))
+		}
+	}
+	return vector.Apply(false, s.eval, vecs...)
+}
+
+func (a *setExpr) eval(in ...vector.Any) vector.Any {
+	offsets, inner := buildList(a.sctx, a.elems, in)
+	// Dedupe list elems
+	vb := vector.NewBuilder(a.sctx.LookupTypeSet(inner.Type()))
+	var b zcode.Builder
+	for i := 0; i < len(offsets)-1; i++ {
```

Comment:
Nit:
```suggestion
	for i := range offsets {
```

Reply:
You can't do this because you don't want traverse the last element in offsets